weldx/utility.py

```diff
@@ -883,3 +939,58 @@ def interp_like(self, da, *args, **kwargs):
 
         """
         return xr_interp_like(self._obj, da, *args, **kwargs)
+
+    def time_ref_unset(self) -> xr.DataArray:
+        """Convert Timedelta + reference Timestamp to DatetimeIndex."""
+        da = self._obj.copy()
+        time_ref = da.weldx.time_ref
+        if time_ref and is_timedelta64_dtype(da.time):
+            da["time"] = da.time.data + time_ref
+            da.time.attrs = self._obj.time.attrs  # restore old attributes !
+        return da
+
+    def time_ref_restore(self) -> xr.DataArray:
+        """Convert DatetimeIndex back to TimedeltaIndex + reference Timestamp."""
+        da = self._obj.copy()
+        time_ref = da.weldx.time_ref
+        if time_ref and is_datetime64_dtype(da.time):
+            da["time"] = pd.DatetimeIndex(da.time.data) - time_ref
+            da.time.attrs = self._obj.time.attrs  # restore old attributes !
+        return da
+
+    def reset_reference_time(self, time_ref_new: pd.Timestamp) -> xr.DataArray:
+        """Return copy with time values shifted to new reference time."""
+        da = self._obj.copy()
+        da = da.weldx.time_ref_restore()
+        da.weldx.time_ref = time_ref_new
+        return da
+
+    @property
+    def time_ref(self) -> pd.Timestamp:
+        """Get the time_ref value or `None` if not set."""
+        da = self._obj
+        if "time" in da.coords:
+            if "time_ref" in da.time.attrs:
+                return da.time.attrs["time_ref"]
+
+        return None
+
+    @time_ref.setter
+    def time_ref(self, value: pd.Timestamp):
+        """Convert INPLACE to new reference time.
+
+        If no reference time exists, the new value will be assigned
+        TODO: should None be allowed and pass through or raise TypeError ?
```

Comment:
Would tend to raise a `ValueError` for now until we think we need to change that

Reply:
Ok, was thinking about if we might need `da1.weldx.time_ref = da2.weldx.time_ref` somewhere